PluginStructs/P248_data_struct.cpp

```diff
@@ -0,0 +1,262 @@
+#ifdef USES_P248
+
+#include "../PluginStructs/P248_data_struct.h"
+
+
+uint8_t AHTSetCalCmd[3]    = { AHTX_CMD, 0x08, 0x00 }; // load factory calibration coeff
+uint8_t AHTSetCycleCmd[3]  = { AHTX_CMD, 0x28, 0x00 }; // enable cycle mode
+uint8_t AHTMeasureCmd[3]   = { 0xAC, 0x33, 0x00 };     // start measurement command
+uint8_t AHTResetCmd        =   0xBA;                   // soft reset command
+
+
+P248_data_struct::P248_data_struct(uint8_t addr) :
+  last_hum_val(0.0f),
+  last_temp_val(0.0f),
+  last_measurement(0),
+  i2cAddress(addr),
+  state(AHT_state::Uninitialized) {}
+
+
+bool P248_data_struct::initialized() const {
+  return state != Uninitialized;
+}
+
+void P248_data_struct::setUninitialized() {
+  state = Uninitialized;
+}
+
+// Only perform the measurements with big interval to prevent the sensor from warming up.
+bool P248_data_struct::update(unsigned long task_index) {
+  const unsigned long current_time = millis();
+
+  if (!initialized()) {
+    if (!begin()) {
+      return false;
+    }
+    state            = Initialized;
+    last_measurement = 0;
+  }
+
+  if (state != Wait_for_samples) {
+    if ((last_measurement != 0) &&
+        !timeOutReached(last_measurement + (Settings.TaskDeviceTimer[task_index] * 1000))) {
+      // Timeout has not yet been reached.
+      return false;
+    }
+
+    last_measurement = current_time;
+
+    startMeasurement();
+    state = Wait_for_samples;
+    return false;
+  }
+
+  //make sure we wait for the measurement to complete
+  if (!timeOutReached(last_measurement + AHT10_MEASURMENT_DELAY)) {
```

Comment:
These measurement delays are < 1000 msec.
Knowing this, is updating via PLUGIN_ONCE_PER_SECOND enough?
Or should you consider PLUGIN_TEN_PER_SECOND?

Reply:
I removed the once per second